skimage/future/trainable_segmentation.py

```diff
@@ -0,0 +1,175 @@
+            label 0 to unlabeled pixels to be segmented.
+        image : ndarray
+            Input image, which can be grayscale or multichannel, and must have a
+            number of dimensions compatible with ``self.features_func``.
+        """
+        if self.features is None:
+            self.compute_features(image)
+        output, clf = fit_segmenter(labels, self.features, self.clf)
+        self.segmented_image = output
+
+    def predict(self, image):
+        """
+        Segment new image using trained internal classifier.
+
+        Parameters
+        ----------
+        image : ndarray
+            Input image, which can be grayscale or multichannel, and must have a
+            number of dimensions compatible with ``self.features_func``.
+
+        Raises
+        ------
+        NotFittedError if ``self.clf`` has not been fitted yet (use ``self.fit``).
+        """
+        if self.features_func is None:
+            self.features_func = multiscale_basic_features
+        features = self.features_func(image)
+        return predict_segmenter(features, self.clf)
+
+
+def fit_segmenter(labels, features, clf):
+    """
+    Segmentation using labeled parts of the image and a classifier.
+
+    Parameters
+    ----------
+    labels : ndarray of ints
+        Image of labels. Labels >= 1 correspond to the training set and
+        label 0 to unlabeled pixels to be segmented.
+    features : ndarray
+        Array of features, with the first dimension corresponding to the number
+        of features, and the other dimensions correspond to ``labels.shape``.
+    clf : classifier object
+        classifier object, exposing a ``fit`` and a ``predict`` method as in
+        scikit-learn's API, for example an instance of
+        ``RandomForestClassifier`` or ``LogisticRegression`` classifier.
+
+    Returns
+    -------
+    clf : classifier object
+        classifier trained on ``labels``
+
+    Raises
+    ------
+    NotFittedError if ``self.clf`` has not been fitted yet (use ``self.fit``).
+    """
+    training_data = features[labels > 0]
+    training_labels = labels[labels > 0].ravel()
+    clf.fit(training_data, training_labels)
+    return clf
+
+
+def predict_segmenter(features, clf):
+    """
+    Segmentation of images using a pretrained classifier.
+
+    Parameters
+    ----------
+    features : ndarray
+        Array of features, with the last dimension corresponding to the number
+        of features, and the other dimensions are compatible with the shape of
+        the image to segment, or a flattened image.
+    clf : classifier object
+        trained classifier object, exposing a ``predict`` method as in
+        scikit-learn's API, for example an instance of
+        ``RandomForestClassifier`` or ``LogisticRegression`` classifier. The
+        classifier must be already trained, for example with
+        :func:`skimage.segmentation.fit_segmenter`.
+    features_func : function, optional
```

Comment:
This parameter is missing in the signature :)

Reply:
oops yes, it's from an old version of the API. Thanks!